components/Input.tsx

```diff
@@ -0,0 +1,58 @@
+import { ChangeEvent, FocusEvent, FormEvent, ReactChild, useState } from 'react';
+import Button from './Button';
+
+interface InputProps {
+  id: string;
+  label: string;
+  name: string;
+  type?: string;
+  placeholder?: string;
+  button?: boolean;
+  buttonLabel?: string;
+  children?: ReactChild;
+  onChange?(inputValue: string): void;
+  onBlur?(inputValue: string): void;
+  onSubmit?(inputValue: string): void;
+}
+
+const Input = (props: InputProps) => {
+  const [isTouched, setIsTouched] = useState(false);
+  const [value, setValue] = useState('');
+
+  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setValue(e.target.value);
+    if (props.onChange && isTouched) {
+      props.onChange(e.target.value);
+    }
+  };
+  const handleBlur = (e: FocusEvent<HTMLInputElement>) => {
+    if (props.onBlur) {
+      props.onBlur(e.target.value);
+    }
+    setIsTouched(true);
+  };
+  const handleSubmit = (e: FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    if (props.onSubmit) {
+      props.onSubmit(value);
+    }
+  };
+
+  return (
+    <form onSubmit={handleSubmit}>
```

Comment:
i wouldnt have a form element be a part of the `Input` component. even if its only temporary. either remove the `form` from the input or create a resuable `Form` in this pull request and update